tests/drivers/interrupt_controller/intc_exti_stm32/src/test_intc_exti_stm32.c

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2025 Alexander Kozhinov <ak.alexander.kozhinov@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/ztest.h>
+#include <zephyr/irq.h>
+#include <zephyr/init.h>
+#include <zephyr/device.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/atomic.h>
+#include <zephyr/drivers/interrupt_controller/intc_exti_stm32.h>
+
+
+#define EXTI_DT_NODE		DT_INST(0, st_stm32_exti)
+#define EXTI_NUM_LINES_TOTAL	DT_PROP(EXTI_DT_NODE, num_lines)
+#define EXTI_NUM_LINES_GPIO	DT_PROP(EXTI_DT_NODE, num_gpio_lines)
+
+#define TEST_EXTI_LINE_NR	DT_PROP(DT_INST(0, test_st_stm32_exti), exti_line_nr)
+#define TEST_EXTI_IRQ_NR	DT_PROP(DT_INST(0, test_st_stm32_exti), exti_line_irq_nr)
+#define TEST_EXTI_IRQ_PRIO	DT_PROP(DT_INST(0, test_st_stm32_exti), exti_line_irq_prio)
+
+static atomic_t is_test_exti_isr_called;
+
+static void test_exti_isr(void)
+{
+	int line_num = -1;
+
+	for (line_num = 0; line_num < EXTI_NUM_LINES_TOTAL; line_num++) {
+		if (stm32_exti_is_pending(line_num)) {
+			stm32_exti_clear_pending(line_num);
+			break;
+		}
+	}
+	zassert_equal(line_num, TEST_EXTI_LINE_NR);
```

Comment:
I don't really see the point of this loop - why should "another EXTI line lower than us is pending" be considered as an error case?

The following is fine by me though.
```suggestion
	zassert_true(stm32_exti_is_pending(TEST_EXTI_LINE_NR));
	stm32_exti_clear_pending(TEST_EXTI_LINE_NR);
```

Reply:
Good point. This loop was introduced to ensure no other line than choosen one is pending. I think it will be better to leave this loop and redesign the test slightly s.t. an assert within the loop occures if line number other than expected is pending.